geocoding_android/android/src/main/java/com/baseflow/geocoding/MethodCallHandlerImpl.java

```diff
@@ -112,17 +112,29 @@ private void onLocationFromAddress(MethodCall call, Result result) {
         }
 
         try {
-            final List<Address> addresses = geocoding.placemarkFromAddress(address);
+            geocoding.placemarkFromAddress(address, new GeocodeListenerAdapter() {
+
+                @Override
+                public void onGeocode(List<Address> addresses) {
+                    if (addresses != null && addresses.size() > 0) {
+                        result.success(AddressMapper.toLocationHashMapList(addresses));
+                    } else {
+                        result.error(
+                                "NOT_FOUND",
+                                String.format("No coordinates found for '%s'", address),
+                                null);
+                    }
+                }
+
+                @Override
+                public void onError(String errorMessage) {
+                    result.error(
+                            "NOT_FOUND",
```

Comment:
Practically these will always be IO errors, so I think it would be better to rename this to `IO_ERROR` to prevent confusion:
```suggestion
                            "IO_ERROR",
```